log4j-core/src/main/java/org/apache/logging/log4j/core/layout/Rfc5424Layout.java

```diff
@@ -570,7 +570,15 @@ private void appendMap(final String prefix, final Map<String, String> map, final
     }
 
     private String escapeSDParams(final String value) {
-        return PARAM_VALUE_ESCAPE_PATTERN.matcher(value).replaceAll("\\\\$0");
+        StringBuilder output = new StringBuilder();
```

Comment:
@adwsingh, could you implement the following changes as well, please?
1. Remove the `PARAM_VALUE_ESCAPE_PATTERN` constant (reorganize the `import`s)
2. Don't instantiate the `StringBuilder` until you encounter one of the target characters. If nothing to be escaped, i.e., `output == null`, you can simply return the `value` as is.
3. Please stick to the coding style followed in the rest of the file; whitespace after `if`, decorate variables with `final` modifier wherever possible, etc.

Reply:
Sure @vy . I wanted to remove `PARAM_VALUE_ESCAPE_PATTERN` but didn't because it was public.

Reply:
@vy: I agree with @adwsingh: unfortunately public constants must be kept (but can be `@Deprecated`), because this is not an internal class.